notebooks/text-generation/mistral-inference-tgi-sagemaker.ipynb

```diff
@@ -0,0 +1,1041 @@
+    "neuronx-distributed==0.6.0\n",
+    "transformers-neuronx==0.9.474"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b839061f-4a4c-494b-810b-8e2e3a179e10",
+   "metadata": {},
+   "source": [
+    "#### Compilation script\n",
+    "This script will be executed by SageMaker. It will download the weights and compile Mistral. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b6b00905-0f32-48e6-8608-50885fc22898",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%writefile src/compile.py\n",
+    "import os\n",
+    "import json\n",
+    "import argparse\n",
+    "from transformers import AutoTokenizer\n",
+    "from optimum.neuron import NeuronModelForCausalLM\n",
+    "\n",
+    "if __name__ == \"__main__\":\n",
+    "    parser = argparse.ArgumentParser()    \n",
+    "    parser.add_argument(\"--compilation_params\", type=str, required=True)\n",
+    "    parser.add_argument(\"--model_id\", type=str, required=False, default=\"yam-peleg/Experiment26-7B\")    \n",
+    "    parser.add_argument(\"--model_dir\", type=str, default=os.environ[\"SM_MODEL_DIR\"])        \n",
+    "    args, _ = parser.parse_known_args()\n",
+    "    \n",
+    "    # parse compilation params\n",
+    "    compilation_params = json.loads(args.compilation_params)\n",
+    "    print(compilation_params)\n",
+    "    # compile and save the model\n",
+    "    model = NeuronModelForCausalLM.from_pretrained(args.model_id, export=True, **compilation_params)\n",
+    "    model.save_pretrained(args.model_dir)\n",
+    "    # now load and export tokenizer\n",
+    "    tokenizer = AutoTokenizer.from_pretrained(args.model_id)\n",
+    "    tokenizer.save_pretrained(args.model_dir)\n",
+    "    print(\"Done!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3c486648-eecf-4355-951b-bf77c602ac48",
+   "metadata": {},
+   "source": [
+    "#### SageMaker Estimator to kick-off model compilation\n",
+    "If you want to deploy your own model, just change the parameter **--model_id** and point it to the correct HuggingFace repo."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4f6238f0-ac95-47f7-8edb-13d9e4f41fb8",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import logging\n",
+    "from sagemaker.pytorch import PyTorch\n",
+    "num_cores=2\n",
+    "seq_len=2048\n",
+    "compilation_params= {\"auto_cast_type\": \"bf16\", \"batch_size\": 1, \"sequence_length\": seq_len, \"num_cores\": num_cores}\n",
+    "model_id='yam-peleg/Experiment26-7B'\n",
+    "\n",
+    "estimator = PyTorch(\n",
+    "    entry_point=\"compile.py\", # Specify your train script\n",
+    "    source_dir=\"src\",\n",
+    "    role=role,\n",
+    "    sagemaker_session=sess,\n",
+    "    container_log_level=logging.DEBUG,\n",
+    "    instance_count=1,\n",
+    "    instance_type='ml.trn1.32xlarge',\n",
```

Comment:
Maybe you could use a smaller instance here for compilation. It is a bit overkill to use such a big host just to export a model on 2 cores.

Reply:
I tried 2xlarge but the job failed due to OOM.